packages/rainbowkit/src/wallets/connectorsForWallets.ts

```diff
@@ -124,6 +124,38 @@ export const connectorsForWallets = (walletList: WalletList) => {
           ...connectionMethods,
         };
 
+        // Initialize i18n support for qrCode steps if exists
+        if (walletInstance.qrCode?.instructions?.steps) {
+          walletInstance.qrCode = {
+            ...walletInstance.qrCode,
+            instructions: {
+              ...walletInstance.qrCode.instructions,
+              steps: walletInstance.qrCode.instructions.steps.map((step) => ({
+                ...step,
+                title: i18n.t(step.title),
+                description: i18n.t(step.title),
+              })),
+            },
+          };
+        }
+
+        // Initialize i18n support for extension steps if exists
+        if (walletInstance.extension?.instructions?.steps) {
+          walletInstance.extension = {
+            ...walletInstance.extension,
+            instructions: {
+              ...walletInstance.extension.instructions,
+              steps: walletInstance.extension.instructions.steps.map(
+                (step) => ({
+                  ...step,
+                  title: i18n.t(step.title),
+                  description: i18n.t(step.title),
+                }),
+              ),
+            },
+          };
+        }
```

Comment:
Pretty striaght forward. If `instructions` exists in `qrCode` or in `extension` we will make sure the steps are localized and translated based off of their string keys: https://github.com/rainbow-me/rainbowkit/pull/1534/files#r1341738637

Reply:
We actually want to use the `i18n` object downstream in the React tree , rather than in `connectorsForWallets`; that way we don't need a refresh or to prepare connectors more than once. Have a look this implementation; we would just use the key from the connector here:
https://github.com/rainbow-me/rainbowkit/blob/252f02e8c024e2f92734653e8ec5fd2180434b07/packages/rainbowkit/src/components/ConnectOptions/ConnectDetails.tsx#L749